lib/antlr4-native/generator.rb

```diff
@@ -147,38 +147,24 @@ class ContextProxy {
           }
 
           Array getChildren() {
-            if (children == nullptr) {
-              children = new Array();
+            Array children;
+            if (orig != nullptr) {
+              for (auto it = orig -> children.begin(); it != orig -> children.end(); it ++) {
+                Object parseTree = ContextProxy::wrapParseTree(*it);
 
-              if (orig != nullptr) {
-                for (auto it = orig -> children.begin(); it != orig -> children.end(); it ++) {
-                  Object parseTree = ContextProxy::wrapParseTree(*it);
-
-                  if (parseTree != Nil) {
-                    children -> push(parseTree);
-                  }
+                if (parseTree != Nil) {
+                  children.push(parseTree);
                 }
               }
             }
-
-            return *children;
+            return children;
           }
 
           Object getParent() {
-            if (parent == Nil) {
-              if (orig != nullptr) {
-                parent = ContextProxy::wrapParseTree(orig -> parent);
-              }
-            }
-
-            return parent;
+            return orig == nullptr ? Nil : ContextProxy::wrapParseTree(orig -> parent);
           }
 
           size_t childCount() {
-            if (orig == nullptr) {
-              return 0;
-            }
-
             return getChildren().size();
```

Comment:
Since we're no longer memoizing the array of children, I think it makes sense to avoid calling `getChildren()` here.
```suggestion
            size_t count = 0;

            for (auto it = orig -> children.begin(); it != orig -> children.end(); it ++) {
              count ++;
            }

            return count;
```

Reply:
Good point, thank you.
        I would then make it even simpler since _orig->children_ is of _std::vector_ type
```        
          size_t childCount() {
            return orig == nullptr ? 0 : orig -> children.size();
          }
```

Reply:
PR is updated